pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix

```diff
@@ -28,27 +32,32 @@ stdenv.mkDerivation rec {
 
   inherit version src;
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake pkg-config ];
 
   buildInputs = [
     indilib libnova curl cfitsio libusb1 zlib boost gsl gpsd
-    libjpeg libgphoto2 libraw libftdi1 libdc1394 ffmpeg fftw
+    libjpeg libgphoto2 libraw.dev libraw.lib libftdi1 libdc1394 ffmpeg fftw
+    limesuite libraw1394 udev
   ] ++ lib.optionals withFirmware [
     firmware
   ];
 
+  patches = [
```

Comment:
If you used `autoPatchelfHook`, the RUNPATH of the precompiled libraries would be automatically fixed (or build error if the dependency cannot be resolved). As a result, you wouldn't need most of the additional buildInputs nor the patches for linking with udev etc.

Reply:
I had the same idea and I've tried with patchelf and autoPatchelfHook. autoPatchelfHook for some reasons unknown to me just skips the problematic Atik and PlayerOne drivers and manually applying patchelf and adding the rpath leads to the obscure situation that `libudev.so.1` and the others are referenced once but not found and appear a **second** time and are found. Tried multiple versions, none worked. :grimacing:

Reply:
Ahhh sorry, you've added it in indi-firmware which actually fixes the issue much more elegantly. Should have checked first. Have changed it according to your suggestion! :)